src/components/BlockMenu.tsx

```diff
@@ -289,13 +283,7 @@ class BlockMenu extends React.Component<Props, State> {
     const parent = findParentNode(node => !!node)(state.selection);
 
     if (parent) {
-      dispatch(
-        state.tr.insertText(
-          "",
-          parent.pos,
-          parent.pos + parent.node.textContent.length + 1
-        )
-      );
+      dispatch(state.tr.insertText("", parent.pos, state.selection.to));
```

Comment:
Replace just the "/slash" text rather than the entire paragraph. This does contain the assumption that the users caret is at the end. If they've used arrow keys to move the caret within the word then half will be left. Because there is no guarantee of a space character separating the search and any existing content I don't see a way around this edgecase.

Reply:
I managed to fix this by using `parent.start` instead of `parent.pos`. Yet, I'm not very sure if this change makes sense or not, so any feedback is welcomed.

Reply:
It's the "end" of the search word that's in question so that shouldn't make a difference from what I can tell.